components/ui/CustomPressable.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { useState } from 'react';
+import { Pressable, Text } from 'react-native';
+
+interface CustomPressableProps {
+  onPress: () => void | Promise<void>;
+  disabled?: boolean;
+  children: React.ReactNode;
+  className?: string;
+}
+
+const CustomPressable: React.FC<CustomPressableProps> = ({
```

Comment:
I like your initiative here but I'd say creating a custom component is jumping the gun for now, as we haven't decided how we want the styling to look. Also we'd probably want to have some kind of animation for the opacity change so it looks nice.

Generally we always want to write less code than needed unless we are absolutely sure, as code itself is a dependency.

Reply:
The only options I could see for using Pressables and keeping visual feedback was to do it this way or write code in every component, and that seemed to be all I could find online. Maybe there's a better solution? 

I thought that for now having this custom component with no specific styling might be OK, so it would maintain the flexibility of TouchableOpacity with the improvements of using Pressable. I guess having an animation on the opacity wouldn't be too tricky

Reply:
It's great you've made a start on solving this, but if it's ok let's put this one in the draft status for now and revisit it when we are getting things ready for native.

Reply:
Sure, I didn't know about draft status. It's pretty much good to go any time I think, so whenever makes sense!